packages/react-components/src/components/RichTextEditor/Plugins/RichTextToolbarPlugin.tsx

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { getFormatState } from 'roosterjs-content-model-api';
+import { getObjectKeys } from 'roosterjs-content-model-dom';
+import type { ContentModelFormatState, EditorPlugin, IEditor, PluginEvent } from 'roosterjs-content-model-types';
+import { PluginEventType } from '../../utils/RichTextEditorUtils';
+
+/**
+ * KeyboardInputPlugin is a plugin for handling keyboard events in the editor.
+ */
+export class RichTextToolbarPlugin implements EditorPlugin {
+  private formatState: ContentModelFormatState | null = null;
+  private editor: IEditor | null = null;
+  onFormatChanged: ((formatState: ContentModelFormatState) => void) | null = null;
+
+  getName(): string {
+    return 'RichTextToolbarPlugin';
+  }
+
+  initialize(editor: IEditor): void {
+    this.editor = editor;
+  }
+
+  dispose(): void {}
```

Comment:
We should dispose the editor here:
this.editor = null;

Reply:
I'll add this in the next PR!